src/index.js

```diff
@@ -118,7 +105,8 @@ function syncReduxAndRouter(history, store, selectRouterState = SELECT_STATE) {
       // trigger an unnecessary `pushState` on load
       lastRoute = initialState;
 
-      store.dispatch(initPath(route.path, route.state));
+      const method = location.action === 'REPLACE' ? replacePath : pushPath;
+      store.dispatch(method(route.path, route.state, { avoidRouterUpdate: true }));
```

Comment:
Here we can just directly use `pushPath` because we are avoiding a route update anyway.